### ISSUE TITLE
`unused-variable` (`N806`) false positive on non-global type aliases

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#7068]
```py
Foo = int | str # no error


def foo():
    Bar = int | str # error: Variable `Bar` in function should be lowercase
```

**Comment by charliermarsh** [comment]
We can probably only support avoiding this when marked as `Bar: TypeAlias` (which I'd recommend).
